pkg/workloads/spark_job/spark_job.py

```diff
@@ -150,6 +151,16 @@ def ingest_raw_dataset(spark, ctx, cols_to_validate, should_ingest):
             logger.info("Ingesting {} data from {}".format(ctx.app["name"], data_config["path"]))
             ingest_df = spark_util.ingest(ctx, spark)
 
+            input_type_map = {f.name: f.dataType for f in ingest_df.schema}
```

Comment:
I moved it here so that the `ingest` can be unit tested without the need to account for metadata upload logic (which I'd have to mock to test properly)